movie_rec/recommender.py

```diff
@@ -1,39 +1,73 @@
 import json
 from optparse import OptionParser
+import psycopg2
 
 class MovieRecommender(object):
-    def __init__(self, hypothesis_path, positive, negative):
+    def __init__(self, hypothesis_path, target_user):
         self.predictions = self.load_hypothesis(hypothesis_path)
-        self.liked_movies = self.load_annotations(positive)
-        self.disliked_movies = self.load_annotations(negative)
+        self.liked_movies = self.load_movies('1', target_user)
+        self.disliked_movies = self.load_movies('-1', target_user)
         self.seen_movies = set(self.liked_movies + self.disliked_movies)
-        
+	self.movie_mapping = self.load_mapping()
+
+    def load_mapping(self):
+        conn = psycopg2.connect('dbname=movie_rec')
+        cur = conn.cursor()
+        query = "select movie_id, title from movies"
+        cur.execute(query)
+        results = cur.fetchall()
+        cur.close()
+        conn.close()
+        return dict([
+            (result[0], result[1])
+            for result in results
+        ])
+
+
+    def load_movies(self, classification, target_user):
+        conn = psycopg2.connect('dbname=movie_rec')
+        cur = conn.cursor()
+        query = """select movie_id from ratings
+                 where user_id = %s and 
+                 rating = %s"""
+        query_data = (target_user, classification)
+        cur.execute(query, query_data)
+        results = cur.fetchall()
+        cur.close()
+        conn.close()
+        return [result[0] for result in results]
+
     def load_hypothesis(self, filename):
         with open(filename) as f:
             return json.loads(f.read())
 
-    def load_annotations(self, filename):
-        with open(filename) as f:
-            return [line.rstrip() for line in f]
-
     def top_n(self, n):
         predict_like = [
             (movie,scores) for movie, scores in self.predictions.items() 
             if scores[1] > scores[0]
         ]
         predict_sorted = sorted(predict_like, key=lambda x: x[1][1], reverse=True)
-        for i in range(n):
-            print i+1, predict_sorted[i][0], predict_sorted[i][1][1]
+        predict_sorted = [x for x in predict_sorted if x[0] not in self.seen_movies]
```

Comment:
Could we just combine this filter with the comprehension above for `predict_like`?